src/main/python/KeyboardToNetworkTables.py

```diff
@@ -0,0 +1,76 @@
+import sys
+import time
+from typing import Callable, TypeVar
+
+from pynput import keyboard
+
+from NetworkTableManager import NetworkTableInstance, NetworkTable, NetworkTableClient
+
+SpecialKey = TypeVar("SpecialKey")
+ASCIIKey = TypeVar("ASCIIKey")
+UndefinedKey = TypeVar("UndefinedKey")
+
+KEYBOARD_EVENT_CHECKING_COOLDOWN_SECONDS = 0.01
+KEYBOARD_KEYS_TABLE = "Keyboard/Keys"
+CLIENT_NAME = "KeyboardToNetworkTables"
+
+
+def key_type_checker(key: object) -> TypeVar:
+    if hasattr(key, "name"):
+        return SpecialKey
+    elif hasattr(key, "char"):
+        return ASCIIKey
+    return UndefinedKey
```

Comment:
@Sh-Pe

Reply:
`hasatter` is one of python's build-in functions/methods. There's no reason to wrap those--it's just as bad as wrapping java's `String` object.

Reply:
when you put String in google translate it says: מחרוזת
when you put hasattr in google translate it doesnt say: בודק את הסוג

so ask Raz what he thinks, but i am against using non informative functions

Reply:
> when you put String in google translate it says: מחרוזת when you put hasattr in google translate it doesnt say: בודק את הסוג
When you put string in google translate it says חוט, and מחרוזת is nowhere to be found among the first 10 words.

Reply:
> when you put String in google translate it says: מחרוזת when you put hasattr in google translate it doesnt say: בודק את הסוג
> 
> so ask Raz what he thinks, but i am against using non informative functions

Also everywhere in the code there are bad names that stuck, `Joysticks` are not the actual joysticks but the controllers, There are constants which aren't constants but objects. In this case this is an operation in python
and wrapping it would be akin to wrapping `&&` to a function called `and`.

Reply:
> > when you put String in google translate it says: מחרוזת when you put hasattr in google translate it doesnt say: בודק את הסוג
> > so ask Raz what he thinks, but i am against using non informative functions
> 
> Also everywhere in the code there are bad names that stuck, `Joysticks` are not the actual joysticks but the controllers, There are constants which aren't constants but objects. In this case this is an operation in python and wrapping it would be akin to wrapping `&&` in a function called `and`.

Not to mention Abbreviations: `SysID`, `PID`, `CTRE`, `CMD` and stuff like `OI`.

Reply:
ik it's resolved but I want to add one thing to it: 
according to the docs (thanks to Emily for pointing it out), it's inherit from a specific, known class, depending on the type (asciikey, specialkey etc.). In particular, I can use `issubclass` or smth in this context, which is more readable. .
Personally I prefer checking each attribute before calling to it, mainly because I don't trust this library to not produce a weird type which will crash the thread. Not a big deal but I think I should lay this out.